Random Data Button

ShmulTomer opened the pull request:
Added a "Random" button for many algorithms to make it easier for students to quickly visualize an example

Added:
- randomized data for all sorts
- LCS: chooses a random pair from a set of examples
- pattern matching: chooses a random pair from a set of examples
- randomized data option for BuildHeap

skapoor68 commented on src/algo/BruteForce.js:
```diff
@@ -139,6 +144,37 @@ export default class BruteForce extends Algorithm {
+			'GGACTGA',
+			'BBBBAABBBAB',
+			'Machine Learning',
+			'Sphinx of black quartz, judge my vow',
```
Similar issue as above but with vertical overflow here. It can be fixed by increasing the canvas height, but that's not ideal.

skapoor68 reviewed (changes requested):
Thanks Tomer this is a really useful feature! I left some comments and let me know if you agree. The main thing is changing the pattern matching examples where the animations get bloated.

ShmulTomer commented:
Thanks Sammy for the feedback!

Changes:
- adjusted examples for pattern matching (shorter, no spaces)
- LSD Radix random button has better values now (more digits compared to the other sorts), also puts negative numbers now too if sort negative numbers is selected
- adjusted “Random” button spacing, added divisor bars 
***I put Random and Clear side by side for pattern matching to maintain the one element height header, but I put them on top of each other for sorting since the header is already larger here; I think it works well this way but lmk your thoughts 
*** I agree "Random Input" is more clear, but when I tried it it just extended the header even further; I think the new way I organized it is cleaner and new users will figure it out quickly but let me know if you disagree

ljones315 commented:
🫡I'll review it later this weekend, thanks for taking a look Sammy!
